activesupport/lib/active_support/file_evented_update_checker.rb

```diff
@@ -0,0 +1,72 @@
+require 'listen'
+
+module ActiveSupport
+  class FileEventedUpdateChecker
+    attr_reader :listener
+    def initialize(files, dirs={}, &block)
+      @files = Set.new
+      @files = files.map { |f| File.expand_path(f)}
+      @dirs = Hash.new
+      dirs.map do |key,value|
+        @dirs[File.expand_path(key)] = Array(value) if !Array(value).empty?
+      end
+      @block = block
+      @modified = false
+      @listener = Listen.to(*base_directories, :latency => 0.1)
```

Comment:
Why set a latency? This class should not be used in polling mode at all.

Nowadays we have code that polls once. That's enough polling: once, not once every something. We want this class only for the evented use case.